### PR TITLE
use poseidon machine in coprocessor substitution

### PR DESCRIPTION
**Pull request by gballet**
Now that the poseidon machine has been merged, this PR connects the `poseidon` instruction with that machine.

This is still WIP, intended to collect some early feedback from @georgwiese and others.

One issue is that running `cargo run --release rust riscv/tests/riscv_data/poseidon.rs -o /tmp` passes, when it really shouldn't.

### CODE REVIEW

File: riscv/src/compiler.rs [hunk 1]
```diff
@@ -115,6 +115,120 @@ machine Shift(latch, function_id) {
         {function_id', A_byte, B', FACTOR_ROW, C_part} in {P_operation, P_A, P_B, P_ROW, P_C};
     }
 }
+"#,
+r#"
```
Comment (gballet) [inline riscv/src/compiler.rs:119]: copied the content of the file, as suggested

File: riscv/runtime/src/coprocessors.rs [hunk 2]
```diff
@@ -4,9 +4,9 @@
 // The function itself will be removed by the compiler
 // during the reachability analysis.
 extern "C" {
-    fn poseidon_coprocessor(a: u32, b: u32) -> u32;
+    fn poseidon_coprocessor(a: u32, b: u32, c: u32) -> u32;
 }
 
 pub fn poseidon_hash(a: u32, b: u32) -> u32 {
-    unsafe { poseidon_coprocessor(a, b) }
+    unsafe { poseidon_coprocessor(a, b, 0) }
```
Comment (gballet) [inline riscv/runtime/src/coprocessors.rs:11]: the capacity is currently set to 0, which I think it wrong. Please advise.

Comment (georgwiese) [reply riscv/runtime/src/coprocessors.rs:11]: In my understanding, 0 is as good as any other constant value. However, the [Poseidon paper](https://eprint.iacr.org/2019/458.pdf) suggests some standard values in Section 4.2, depending on the use-case.

**Comment by leonardoalt** [comment]
done in https://github.com/powdr-labs/powdr/pull/678

**Pull request closed without merging**